ui/apps/platform/cypress/helpers/vulnmanagement/entities.js

```diff
@@ -41,40 +42,268 @@ if (hasFeatureFlag('ROX_FRONTEND_VM_UPDATES')) {
     );
 }
 
-export function visitVulnerabilityManagementDashboardFromLeftNav() {
-    opnamesForDashboard.forEach((opname) => {
-        cy.intercept('POST', api.graphql(opname)).as(opname);
+/*
+ * For example, given ['searchOptions', 'getDeployments'] return:
+ * {
+ *     searchOptions: '/api/graphql?opname=searchOptions',
+ *     getDeployments: '/api/graphql?opname=getDeployments',
+ * }
+ */
+function routeMatcherMapForOpnames(opnames) {
+    const routeMatcherMap = {};
+
+    opnames.forEach((opname) => {
+        routeMatcherMap[opname] = api.graphql(opname);
     });
 
-    visitFromLeftNavExpandable('Vulnerability Management', 'Dashboard');
+    return routeMatcherMap;
+}
+
+const requestConfigForDashboard = {
+    routeMatcherMap: routeMatcherMapForOpnames(opnamesForDashboard),
+};
+
+export function visitVulnerabilityManagementDashboardFromLeftNav() {
+    visitFromLeftNavExpandable('Vulnerability Management', 'Dashboard', requestConfigForDashboard);
 
-    cy.wait(opnamesForDashboard.map((opname) => `@${opname}`));
     cy.get('h1:contains("Vulnerability Management")');
 }
 
 export function visitVulnerabilityManagementDashboard() {
-    opnamesForDashboard.forEach((opname) => {
-        cy.intercept('POST', api.graphql(opname)).as(opname);
-    });
-
-    visit(url.dashboard);
+    visit(url.dashboard, requestConfigForDashboard);
 
-    cy.wait(opnamesForDashboard.map((opname) => `@${opname}`));
     cy.get('h1:contains("Vulnerability Management")');
 }
 
+const { opnameForEntity } = api; // TODO move here from apiEndpoints.js
+
+const { opnameForEntities } = api; // TODO move here from apiEndpoints.js
+
+const { opnamePrefixForPrimaryAndSecondaryEntities } = api; // TODO move here from apiEndpoints.js
+
+const keyForEntity = {
+    clusters: 'CLUSTER',
+    components: 'COMPONENT',
+    'image-components': 'IMAGE_COMPONENT',
+    'node-components': 'NODE_COMPONENT',
+    cves: 'CVE',
+    'image-cves': 'IMAGE_CVE',
+    'node-cves': 'NODE_CVE',
+    'cluster-cves': 'CLUSTER_CVE',
+    deployments: 'DEPLOYMENT',
+    images: 'IMAGE',
+    namespaces: 'NAMESPACE',
+    nodes: 'NODE',
+    policies: 'POLICY',
+};
```

Comment:
Fine where it is for this PR, but this looks like something that belongs exported from a helper file. It could be used in multiple VM tests files, and even in Config Mgmt tests.